### ISSUE TITLE
Tensorflow looks for wrong libcupti.so version

### ISSUE THREAD

**Issue by StefanoD** [tensorflow/tensorflow#9341]
- OS: Ubuntu 16.04
- *TensorFlow installed from: source
- *TensorFlow version*: 1.1.0-rc2
- *Bazel version (if compiling from source)*: 0.4.5
- *CUDA/cuDNN version*: 8.0/6.0
- *GPU Model and Memory*: NVIDIA GTX 1060, 6 GB RAM
- *Exact command to reproduce*:
------------------------
MNIST examples works fine, bute MNIST example with summaries crashes with:

```
(tensorflow) stefano@stefano-linux:~/Dokumente/Programming/Python/TensorflowCoreTutorial$ /home/stefano/tensorflow/bin/python3 /home/stefano/Dokumente/Programming/Python/TensorflowCoreTutorial/src/mnist_with_summaries.py
Extracting /home/stefano/Dokumente/Programming/Python/MNIST/train-images-idx3-ubyte.gz
Extracting /home/stefano/Dokumente/Programming/Python/MNIST/train-labels-idx1-ubyte.gz
Extracting /home/stefano/Dokumente/Programming/Python/MNIST/t10k-images-idx3-ubyte.gz
Extracting /home/stefano/Dokumente/Programming/Python/MNIST/t10k-labels-idx1-ubyte.gz
2017-04-20 19:16:15.388639: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:901] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2017-04-20 19:16:15.389019: I tensorflow/core/common_runtime/gpu/gpu_device.cc:887] Found device 0 with properties:
name: GeForce GTX 1060 6GB
major: 6 minor: 1 memoryClockRate (GHz) 1.7085
pciBusID 0000:22:00.0
Total memory: 5.93GiB
Free memory: 4.97GiB
2017-04-20 19:16:15.389036: I tensorflow/core/common_runtime/gpu/gpu_device.cc:908] DMA: 0
2017-04-20 19:16:15.389042: I tensorflow/core/common_runtime/gpu/gpu_device.cc:918] 0:   Y
2017-04-20 19:16:15.389055: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 1060 6GB, pci bus id: 0000:22:00.0)
Accuracy at step 0: 0.0981
Accuracy at step 10: 0.7258
Accuracy at step 20: 0.8151
Accuracy at step 30: 0.8572
Accuracy at step 40: 0.879
Accuracy at step 50: 0.8949
Accuracy at step 60: 0.9052
Accuracy at step 70: 0.9058
Accuracy at step 80: 0.915
Accuracy at step 90: 0.9137
2017-04-20 19:16:20.708545: I tensorflow/stream_executor/dso_loader.cc:129] Couldn't open CUDA library libcupti.so.8.0. LD_LIBRARY_PATH: /usr/local/lib:/usr/local/cuda/lib64:/usr/lib/x86_64-linux-gnu:/usr/lib/nvidia-375
2017-04-20 19:16:20.708600: F tensorflow/core/platform/default/gpu/cupti_wrapper.cc:59] Non-OK-status: ::tensorflow::Env::Default()->GetSymbolFromLibrary( GetDsoHandle(), kName, &f) status: Not found: /usr/local/lib/python3.5/dist-packages/tensorfl
ow/python/_pywrap_tensorflow_internal.so: undefined symbol: cuptiActivityRegisterCallbackscould not find cuptiActivityRegisterCallbacksin libcupti DSO
```

source code:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/mnist/mnist_with_summaries.py

Problem is, that tensorflow looks for libcupti.so.8.0, but only libcupti.so.7.5 is installed:

```
stefano@stefano-linux:~$ ls /usr/lib/x86_64-linux-gnu | grep libcupti*
libcupti.so
libcupti.so.7.5
libcupti.so.7.5.18
```

I installed libcupti like described in the Install section on the Tensorflow website:

```
sudo apt-get install libcupti-dev
```

libcupti-dev 8.0 is not available for Ubuntu 16.04.

**Comment by drpngx** [comment]
It looks for `DSO_CUDA_VERSION` which is the one you built it for.

**Comment by StefanoD** [comment]
Yes, I build it for CUDA 8.0. CUDA 8.0 is available for Ubuntu 16.04, but libcupti-dev is only available in version 7.5 for Ubuntu 16.04.

Do I have to install libcupti-dev 8.0 somehow?

**Comment by drpngx** [comment]
@yifeif @jart we set the `DSO_CUDA_VERSION` from the CUDA lib, but apparently the profiler tools can have a different version. (the `DSO_CUDA_VERSION` is used to load both cuda lib and cupti). Since ubuntu 16.04 is the primary version for us, it should work out of the box. Should we `configure` to split the versions?

**Comment by StefanoD** [comment]
For everyone who has the same issue:

First download the libcupti8.0 debian package and install it:
http://packages.ubuntu.com/yakkety/amd64/libcupti8.0/download

Afterwards download libcupti-dev8.0 debian package in order to install it:
https://ubuntu.pkgs.org/16.10/ubuntu-multiverse-amd64/libcupti-dev_8.0.44-0ubuntu1_amd64.deb.html

This should solve this issue. Of course it would be still better if everything would work out of the box.

**Comment by jupiter126** [comment]
Would anyone happen to know where someone not running a Debian derivative can get the source for libcupti?

**Comment by gunan** [comment]
Looks like @StefanoD was able to find the resolution to this issue.

For posterity, You have to have the same version of CUDA and libcupti on your system. Latest version of libcupti or cuda may not be on debian repositories yet, so you may need to download these from nvidia, or more recent ubuntu distribution's repositories.

**Comment by bzamecnik** [comment]
`libcupti` is available in Ubuntu 16.04 xenial only for CUDA 7.5, not CUDA 8. We can use deb packages from Ubuntu zesty repo:

```
wget http://cz.archive.ubuntu.com/ubuntu/pool/multiverse/n/nvidia-cuda-toolkit/libcupti8.0_8.0.44-3_amd64.deb
wget http://cz.archive.ubuntu.com/ubuntu/pool/multiverse/n/nvidia-cuda-toolkit/libcupti-dev_8.0.44-3_amd64.deb
sudo dpkg -i libcupti8.0_8.0.44-3_amd64.deb libcupti-dev_8.0.44-3_amd64.deb
```

It can be tested eg. by the scripts in https://medium.com/towards-data-science/howto-profile-tensorflow-1a49fb18073d.

```
[...]
2017-08-22 16:17:42.714397: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1045] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX 980 Ti, pci bus id: 0000:01:00.0)
2017-08-22 16:17:42.740833: I tensorflow/stream_executor/dso_loader.cc:139] successfully opened CUDA library libcupti.so.8.0 locally
```

**Comment by ahundt** [comment]
Just adding another error string so this will turn up in search for people in the future:

```
2017-12-20 18:18:50.156691: F ./tensorflow/stream_executor/lib/statusor.h:212] Non-OK-status: status_ status: Failed precondition: could not dlopen DSO: libcupti.so.8.0; dlerror: libcupti.so.8.0: cannot open shared object file: No such file or directory
```
